meshapp/src/components/login-form.tsx

```diff
@@ -22,6 +26,35 @@ const LoginScreen = (props: ComponentProps) => {
     console.log(formData);
   };
 
+  const navigate = useNavigate();
+  const [cookies, setCookie] = useCookies(['user_id']);
+
+  useEffect(() =>{},[cookies]);
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const response = await axiosInstance.post('accounts/login/',{
+      "email":formData.user,
+      "password":formData.pass
+    })
+      .then((axiosResponse) => {
+        if(axiosResponse.data.enabled_2fa){
+          setCookie('user_id', axiosResponse.data.user_id, { path:'/', maxAge:150 });
+          axiosInstance.post('accounts/set-two-factor-auth/', {
+            "accountID":axiosResponse.data.user_id
+          })
+          .then(() => {
+            navigate('/otp');
+          })
+          .catch(errorHandler);  // Somehow, user's account ID could not be found
+        }
+        else{
+          navigate('/logged_in_home')
+        }
+      })
+      .catch(errorHandler);  // User attempted to login and an error occured
+  }
```

Comment:
Looking back on these lines of code, I wonder if using try/catch along with await in more places instead of .then, .catch would be a bit more readable and doesn't require directly responding to promises.

Reply:
Good job!